### DISCUSSION TITLE
[Question] Is it possible to nest a PDF file?

### DISCUSSION THREAD

**Discussion by HZSamir** [diegomura/react-pdf#2175] [Q&A] [answered]
Hello,
Allow me to apologize for creating this issue: Turns out using the keyword PDF to search the issues of a PDF-related lib is a bad idea.
I'd like to be able to nest an existing PDF file inside my new created PDF, and was wondering if that was at all possible?
Maybe natively in this lib, or maybe by combining wojtekmaj/react-pdf?
I'd just like to know if that's possible, and if so, how?

Thank you very much for your time.

**Comment by jeetiss** [comment, marked as answer]
You can use this library to generate new pages and combine them with existing PDF using [pdf-merger-js](https://www.npmjs.com/package/pdf-merger-js) or [pdf-lib](https://www.npmjs.com/package/pdf-lib)

**Reply by HZSamir** [reply]
Thank you. I'll give it a try.
Good day to you.

**Comment by diegomura** [comment]
Hi! No, this is not supported. react-pdf only generates new PDF documents. It cannot parse an existing PDF file, so there is no way to nest one inside your document.

Two common workarounds:

1. Convert the pages of the existing PDF to images and place them with the `Image` component.
2. Generate your document with react-pdf first, then merge it with the existing PDF using a library like [pdf-lib](https://github.com/Hopding/pdf-lib) as a second step.

Closing this one since embedding existing PDFs is out of scope for this library. Thanks!
